src/services/apiService.js

```diff
@@ -0,0 +1,73 @@
+export const apiService = {
+  BASE_API_URL: 'https://blue-bottle-api-test.herokuapp.com',
+  GET_COFFEE_SHOPS: '/v1/coffee_shops',
+  GET_TOKEN: '/v1/tokens',
+
+  async getNearestShops(position) {
+    const coffeeShops = await this.getCoffeeShops();
+    return coffeeShops.map(coffeeShop => {
+      return {
+        id: coffeeShop.id,
+        name: coffeeShop.name,
+        latitude: parseFloat(coffeeShop.x),
+        longitude: parseFloat(coffeeShop.y)
+      }
+    });
+  },
+
+  async getCoffeeShops(apiKey = '') {
+    if (apiKey === '') {
+      apiKey = await this.retrieveNewToken();
+    }
+
+    console.log("Getting the coffee shops...");
```

Comment:
Console logs no longer make sense in this situation, it made sense for a console app. You can use the state of the request to check if it is still loading or it gave an error and display the corresponding HTML. (But this is a bit more advanced so leave it for now, as it is just nice to have)